### PR TITLE
Use JsVal instead of JsRef in more places (nr. 3)

### PR DESCRIPTION
**Pull request by hoodmane**
This also fixes `JsArray_pop` which leaked memory and returned a singleton list with the element removed rather than the element itself.

- [x] Add a [CHANGELOG](https://github.com/pyodide/pyodide/blob/main/docs/project/changelog.md) entry
- [x] Add / update tests

### CODE REVIEW

File: src/tests/test_jsproxy.py [hunk 1]
```diff
@@ -105,16 +105,16 @@ def test_jsproxy_document(selenium):
 @pytest.mark.parametrize(
     "js,result",
     [
-        ("{}", False),
+        ("{}", True),
```
Comment (ryanking13) [inline src/tests/test_jsproxy.py:108]: Were we treating an empty JS object as a falsy value before?
I got a different result when I tried this.

```js
a = {}
pyodide.runPython(`
from js import a
print(bool(a))
`)
```
```
True
```

**Review by ryanking13 — APPROVED** [review]
Thanks! LGTM with a question.

> This also fixes JsArray_pop which leaked memory and returned a singleton list with the element removed rather than the element itself.

Nice!

File: src/tests/test_jsproxy.py (hunk 1 above)
Comment (hoodmane) [reply src/tests/test_jsproxy.py:108]: Empty object was always truthy, but the test was written incorrectly. If you do `eval("{}")`, `{}` is parsed as an empty block not as an object. Thus, it returns `undefined` which is `Falsey`. I adjusted the test so it calls `run_js(f"({js})")` which calls `eval("({})")` which returns empty object with is truthy.

I think empty object is most comparable to:
```py
class A:
   pass
bool(A())
```
We could do something like:
```js
function isTruthy(obj) {
  if (obj.constructor !== Object && obj.constructor !== null) {
    return true;
  }
  for (const prop in obj) {
    if (Object.hasOwn(obj, prop)) {
      return true;
    }
  }
  // object with default or null constructor and no ownProperties
  return false;
}
```

Comment (hoodmane) [reply src/tests/test_jsproxy.py:108]: Anyways I am open to switching to this in the future.

**Comment by hoodmane** [comment]
I added a lot of methods in the same PR with `JsArray_pop`, so it stands to reason that I forgot to test one.

File: src/tests/test_jsproxy.py (hunk 1 above)
Comment (ryanking13) [reply src/tests/test_jsproxy.py:108]: I see. Thanks for the explanation!

**Pull request merged**